# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../conntrack-tools/0003-include-sys-select.h.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

include sys/select.h to get fd_set

--- ./src/udp.c.orig	2013-09-01 01:47:48.965734257 +0200
+++ ./src/udp.c	2013-09-01 01:49:11.671609969 +0200
@@ -7,8 +7,6 @@
  * (at your option) any later version.
  */
 
-#include "udp.h"
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <arpa/inet.h>
@@ -18,6 +16,9 @@
 #include <net/if.h>
 #include <errno.h>
 #include <limits.h>
+#include <sys/select.h>
+
+#include "udp.h"
 
 struct udp_sock *udp_server_create(struct udp_conf *conf)
 {
--- ./src/tcp.c.orig	2013-09-01 02:22:00.461217706 +0200
+++ ./src/tcp.c	2013-09-01 02:24:13.520277224 +0200
@@ -9,8 +9,6 @@
  * TCP support has been sponsored by 6WIND <www.6wind.com>.
  */
 
-#include "tcp.h"
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <arpa/inet.h>
@@ -21,7 +19,9 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <limits.h>
+#include <sys/select.h>
 
+#include "tcp.h"
 #include "conntrackd.h"
 #include "fds.h"
 
--- ./include/mcast.h.orig	2013-09-01 02:42:11.601231152 +0200
+++ ./include/mcast.h	2013-09-01 02:42:57.838277721 +0200
@@ -4,6 +4,7 @@
 #include <stdint.h>
 #include <netinet/in.h>
 #include <net/if.h>
+#include <sys/select.h> /* for fd_set */
 
 struct mcast_conf {
 	int ipproto;
